File: articles/regret-analysis-and-minimax-decision-rules/r/run_all_regret_minimax_workflows.R
# run_all_regret_minimax_workflows.R
# Run all R workflows for Regret Analysis and Minimax Decision Rules.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "regret_analysis_minimax_decision_rules_workflow.R",
  "regret_tables.R",
  "minimax_rule_tables.R",
  "maximin_rule_tables.R",
  "threshold_review_tables.R",
  "decision_rule_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R regret and minimax workflows completed.")

File: articles/regret-analysis-and-minimax-decision-rules/python/run_all_regret_minimax_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Regret Analysis and Minimax Decision Rules."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "payoff_matrix_builder.py",
    "regret_matrix_calculator.py",
    "maximin_rule.py",
    "minimax_regret_rule.py",
    "threshold_compliance.py",
    "vulnerability_analysis.py",
    "decision_rule_comparison.py",
    "decision_record_exporter.py",
    "regret_analysis_minimax_decision_rules_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python regret and minimax workflows completed.")


if __name__ == "__main__":
    main()
